pkg/client/vault/client.go

```diff
@@ -205,3 +343,59 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, error) {
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(lastElm(keyURL, "/"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, nil
+}
+
+func encryptContent(wKMS KeyManager, wCrypto crypto.Crypto, content interface{}) (string, error) {
+	src, err := json.Marshal(content)
+	if err != nil {
+		return "", fmt.Errorf("marshal: %w", err)
+	}
+
+	_, kidURL, err := wKMS.Create(kms.NISTP256ECDHKW)
```

Comment:
There should be no need to create an asymmetric key for encryption. In this case it results in additional network overhead. The encryption key should be created locally.